packages/components/src/SquigglePlayground.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC, useState } from "react";
+import ReactDOM from "react-dom";
+import { SquiggleChart } from "./SquiggleChart";
+import CodeEditor from "./CodeEditor";
+import { Form, Input, Card, Row, Col } from "antd";
+import "antd/dist/antd.css";
+
+interface FieldFloatProps {
+  label: string;
+  className?: string;
+  value: number;
+  onChange: (value: number) => void;
+}
+
+function FieldFloat(Props: FieldFloatProps) {
+  let [contents, setContents] = useState(Props.value + "");
+  return (
+    <Form.Item label={Props.label}>
+      <Input
+        value={contents}
+        className={Props.className ? Props.className : ""}
+        onChange={(e) => setContents(e.target.value)}
+        onBlur={(_) => {
+          let result = parseFloat(contents);
+          if (result != NaN) {
+            Props.onChange(result);
+          }
+        }}
+      />
+    </Form.Item>
+  );
+}
+
+interface Props {
+  initialSquiggleString: string;
+}
+
+let SquigglePlayground: FC<Props> = (props) => {
+  let [squiggleString, setSquiggleString] = useState(
+    props.initialSquiggleString
+  );
+  let [sampleCount, setSampleCount] = useState(1000);
```

Comment:
I'd probably take out these variables into a ``defaults`` hash. Like,

```
const formDefaults = {
  sampleCount: 1000,
...
```

Having "magic" values in the code base is an anti-pattern.

Reply:
I imagine we could use ``defaultProps`` for this?
https://reactjs.org/docs/typechecking-with-proptypes.html
https://blog.logrocket.com/a-complete-guide-to-default-props-in-react-984ea8e6972d/